src/helics/application_api/ValueFederate.cpp

```diff
@@ -319,30 +305,13 @@ void ValueFederate::registerValueInterfacesToml (const std::string &tomlString)
             {
                 continue;
             }
-            auto units = tomlGetOrDefault (sub, "units", std::string ());
-            bool optional = tomlGetOrDefault (sub, "optional", false);
-            auto info = tomlGetOrDefault (sub, "info", std::string ());
-            bool required = tomlGetOrDefault (sub, "required", !optional);
-            id = &registerSubscription (key, units);
-            if (required)
-            {
-                // setInterfaceOption()
-            }
+            auto type = getOrDefault (sub, "type", emptyStr);
+            auto units = getOrDefault (sub, "units", emptyStr);
 
-            auto shortcut = sub.find ("shortcut");
-            if (shortcut != nullptr)
-            {
-                addShortcut (*id, shortcut->as<std::string> ());
-            }
-            auto tol = tomlGetOrDefault (sub, "tolerance", -1.0);
-            if (tol > 0.0)
-            {
-                id->setMinimumChange (tol);
-            }
-            if (!info.empty ())
-            {
-                setInfo (id->getHandle (), info);
-            }
+            id = &registerInput (emptyStr, type, units);
+            id->addTarget (key);
```

Comment:
the updated subscription registration from #428, allowing subscriptions to have types from the config files